collect_app/src/main/java/org/odk/collect/android/provider/InstanceProvider.java

```diff
@@ -385,6 +385,12 @@ public int update(@NonNull Uri uri, ContentValues values, String where, String[]
         return count;
     }
 
+    private boolean isDatabaseHelperOutOfDate() {
```

Comment:
I'm generally fine with static helpers inside a class - especially for the reasoning that it makes them easier to move. I've even found it very useful in heavier objects to even use static methods that just take in instance fields as parameters as sometimes this can end up more readable. They can be problematic when you realize that a static method is a dependency from some object you'd like to test but usually it's pretty easy to then create an interface where that method should live - this was done recently with the `PermissionUtils`.

In this case it makes more sense to me to move the helper to `InstanceDatabaseHelper` as a public method. For me then you're just asking the object "are you out of date?" and then the `InstancesProvider` can decide what to do with it. What would be really nice would be to follow the ["tell don't ask"](https://www.martinfowler.com/bliki/TellDontAsk.html) principle here and call something on `InstancesDatabaseHelper` that makes it upgrade itself if needed. I don't think that's possible though... 🤷‍♂️

Reply:
Done.